### PR TITLE
Fix KeyError in eclext analyser

### PR DESCRIPTION
**Pull request by flacombe**
Normally fixes the KeyError when accessing to dicts.

No more syntax errors at home.

But tests shows the following, why?

```
2023-05-20 23:06:21   update
2023-05-20 23:06:21     iteration=1
2023-05-20 23:06:28       error: UPDATE ERROR france_rhone_alpes_haute_savoie/Merge_Eclext_FR_Syane : {"detail":"AUTH FAIL"}

2023-05-20 23:06:43     iteration=2
2023-05-20 23:06:49       error: UPDATE ERROR france_rhone_alpes_haute_savoie/Merge_Eclext_FR_Syane : {"detail":"AUTH FAIL"}

2023-05-20 23:07:19     iteration=3
2023-05-20 23:07:26       error: UPDATE ERROR france_rhone_alpes_haute_savoie/Merge_Eclext_FR_Syane : {"detail":"AUTH FAIL"}
```

### CODE REVIEW

File: analysers/Analyser_Merge_street_lamp_FR_eclext.py [hunk 1]
```diff
@@ -50,16 +50,16 @@ def __init__(self, config, source_url, dataset_name, source, srid, osmRef, class
...
+                        'light:method': lambda res: self.extract_light_method.get(res['typeSource']),
+                        'light:colour': lambda res: self.extract_temperature(res['TemperatureCouleur']),
+                        'light:flux': lambda res: self.clean_float(res.get('fluxSource')) if res.get('fluxSource') else None,
+                        'light:height': lambda res: self.clean_float(res.get('hauteurFeu')) if res.get('hauteurFeu') else None,
+                        'light:power': lambda res: self.clean_float(res['puissance']),
+                        'light:lit': lambda res: self.extract_adaptatif.get(res.get('eclairageAdaptatif')) if res.get('eclairageAdaptatif') else None,
+                        'support': lambda res: self.extract_support.get(res['support']),
+                        'manufacturer': lambda res: self.clean_string_tag(res.get('fabricant')) if res.get('fabricant') else None,
+                        'model': lambda res: self.clean_string_tag(res.get('model')) if res.get('model') else None,
+                        'power': lambda res: 'pole' if (res['typeSource'] == 'POT') else None,
```
Comment (frodrigo) [inline analysers/Analyser_Merge_street_lamp_FR_eclext.py:62]: You are sure about this change ?

**Comment by frodrigo** [comment]
> But tests shows the following, why?

Because it works, it try to upload the result.

File: analysers/Analyser_Merge_street_lamp_FR_eclext.py (hunk 1 above)
Comment (flacombe) [reply analysers/Analyser_Merge_street_lamp_FR_eclext.py:62]: I've check the mandatory status of each field in the standard.

If mandatory => []
If optional => .get()

Invalid files should indeed raise errors and be refined by producers

**Comment by flacombe** [comment]
> Because it works, it try to upload the result.

Weird, I didn't get this result before.
By the way, good.

**Comment by frodrigo** [comment]
Thank you. Merged, but not deployed yet.

**Comment by frodrigo** [comment]
It is ok now

http://osmose.openstreetmap.fr/fr/map/#source=441424&item=8490&class=1001&zoom=17&lat=46.107666&lon=6.266214&level=1%2C2%2C3

**Comment by flacombe** [comment]
Sounds good, thank you!

**Pull request merged**